### DISCUSSION TITLE
Pylon package, OpenCV.Net.Native.NativeMethods

### DISCUSSION THREAD

**Discussion by GerrikLabra** [bonsai-rx/bonsai#1639] [General]
Hey, could anyone help me solve an issue with the Pylon package on Bonsai? I am trying to setup a Baseler Camera to capture images through bonsai, but its dependent package, OpenCV, is throwing an error. "The type initializer for 'OpenCV.Net.Native.NativeMethods' threw an exception. file 'streamgrabberimpl.cpp', line 469". This error has been consistent during my attempts to fix it.

This has been a recurring issue for the pylon package, with several threads already on this site. However, the last one was 2021, and it's solution was ineffective for my situation. So a new thread is warranted.

### Hardware 
Hardware is Baseler cameras, GigE ethernet connections to a POE Injector, connected to a router, router connected to internet and computer.
Bonsai Editor v.2.8.0 Clean installed

###  Software
Pylon Viewer 64 Bit and Pylon Runtime.
- Release 7.10.25066
- c++ SDK 7.1.0.19126
- App 1.3.1.11271
- Viewer 7.1.0.11271

Bonsai- pylon library
created by Goncalo Lopes
V0.30
Published 11/29/2023
OpenCV.Net[3.3.1)

PFS file generated correctly, serial number matches PFS.
Workflow example:

Pylon Capture->Image->VideoWriter

### Hunting Error
Attempted to find the exact line of code, found out that OpenCV doesnt support GigE according to this [doc](https://www2.baslerweb.com/media/downloads/documents/application_notes/AW00136803000_Getting_Started_with_pylon_6_and_OpenCV.pdf), then the OpenCV [github](https://github.com/horizongir/opencv.net/tree/main/src/OpenCV.Net.UnitTests) doesn't have a file "OpenCV.Net.Native.NativeMethods".

### Attempts to fix
Did multiple clean downloads of Bonsai, Pylon Viewer, opencv-4.8.1-windows.exe, vcredist_x64.exe, and visualstudiosetup.exe.

Chaning video capture settings, image acquisition is set to Mono 8 or 12.

Changed grab strategy.

Switched cameras

Direct connection from computer to camera

Asked public ChatGPT, but its out of date from current Bonsai.

**Comment by vigji** [comment]
Any update on this? I saw there's also #1553 around the same issue, but none of the proposed solutions worked for me (I mount Pylon 7.1.0, nuked Bonsai multiple times, installed 2.8.2, removed all nuget cached packages, tried out the Visual C++ Redistributable for Visual Studio 2012 Update 4 code @bruno-f-cruz was suggesting in that other issue , but no luck)

**Reply by bruno-f-cruz** [reply]
I just can't reproduce the issue on my side so I am not sure how I can help :( ... At this point, I am wondering if it is a specific problem with the drivers Basler uses for GigaE cameras, which model are you using?

**Reply by vigji** [reply]
Basler ace, acA1440-220um USB 3.0 interface...funnily enough, exactly same camera on another machine of a different setup with the same Pylon SDK and Bonsai version works just fine, I have low reproducibility of the issue myself :D

Not sure what could be good pointers to dig further...

**Reply by vigji** [reply]
Came back with a clearer mind, turned out I was keep installing the Pylon runtime interface instead of the developer SDK :) solved it for me!

**Reply by bruno-f-cruz** [reply]
That is an interesting data point. We will keep it in mind. Thanks for reporting it!

**Reply by vigji** [reply]
Additional data point: after managing to having it working, a colleague re-installed the Pylon runtime interface stand-alone and it caused the same disruption. To solve it, I had to uninstall again the runtime interface.
